app/rules/RuleProvisionerService.scala

```diff
@@ -0,0 +1,44 @@
+package rules
+
+import java.util.Date
+
+import akka.actor.Scheduler
+import matchers.RegexMatcher
+import services.MatcherPool
+import utils.Loggable
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class RuleProvisionerService(ruleBucket: BucketRuleResource, matcherPool: MatcherPool)(implicit ec: ExecutionContext) extends Loggable with Runnable {
```

Comment:
More a learning point for me: why do we need `Runnable`? We're already passing in the scheduler.

Reply:
The reason for this was that `scheduler.scheduleWithFixedDelay` takes a Runnable object as a parameter. The `scheduler.schedule` method can take a function instead but is now deprecated. https://doc.akka.io/japi/akka/current/akka/actor/Scheduler.html

Reply:
Ah, yep, just spotted the `this` and the `override def run` 😀